src/loadConfigurationFile.js

```diff
@@ -1,83 +1,93 @@
             return 1;
         }
         return 0;
     }
     return res;
-});
+};
+
+availableExts.sort(compareExtensions);
 
-function getMatchingLoaderFn(configPath, extensions, variants) {
-    let availableExtensions =  extensions || availableExts;
+const getMatchingLoaderFn = (configPath, extensions, variants) => {
+    let availableExtensions = extensions || availableExts;
     let jsVariants = variants || jsVars;
-    for(var i = 0, len = availableExtensions.length; i < len; i++) {
-        var ext = availableExtensions[i];
-        if(endsWith(configPath, ext)) {
-            return jsVariants[ext];
-        }
-    }
-    return null;
-}
+    let retVal = null;
 
-function callConfigFunction(fn) {
-    return fn(require('minimist')(process.argv, { '--': true }).env || {});
-}
+    availableExtensions.some(ext => {
+        if (endsWith(configPath, ext)) {
+            retVal = jsVariants[ext];
+            return true;
+        }
+    });
+    return retVal;
+};
 
-function getConfig(configPath) {
-    var configModule = require(configPath);
-    var configDefault = configModule && configModule.__esModule ? configModule.default : configModule;
-    return typeof configDefault === 'function' ? callConfigFunction(configDefault) : configDefault;
-}
+const callConfigFunction = fn =>
+    fn(require('minimist')(process.argv, { '--': true }).env || {});
 
-module.exports = {
-    default: function(configPath, matchingLoader) {
-        let getMatchingLoader = matchingLoader || getMatchingLoaderFn;
+const getConfig = configPath => {
+    const configModule = require(configPath);
+    const configDefault =
+        configModule && configModule.__esModule
+            ? configModule.default
+            : configModule;
+    return typeof configDefault === 'function'
+        ? callConfigFunction(configDefault)
+        : configDefault;
+};
 
-        var mod = getMatchingLoader(configPath);
-        if(mod) {
-            var mods = Array.isArray(mod) ? mod : [mod],
-                installed = false;
+export default (configPath, matchingLoader) => {
+    const getMatchingLoader = matchingLoader || getMatchingLoaderFn;
 
-            for(var i = 0, len = mods.length; i < len; i++) {
-                mod = mods[i];
-                if(typeof mod === 'string') {
-                    try {
-                        require(mod);
-                        installed = true;
-                    } catch(ignored) {}
-                } else if(typeof mod === 'object') {
-                    try {
-                        var s = require(mod.module);
-                        mod.register(s);
-                        installed = true;
-                    } catch(ignored) {}
-                }
+    let mod = getMatchingLoader(configPath);
+    if (mod) {
+        let mods = Array.isArray(mod) ? mod : [mod];
+        let installed = false;
 
-                if(installed) {
-                    break;
-                }
+        for (let mod of mods) {
```

Comment:
This would be a good case for `.some`:

```
const installed = mods.some(mod => {
    if (typeof mod === 'string') {
        try {
             require(mod);
             return true;
        } catch (ignored) {}
    } else if (typeof mod === 'object') {
        try {
            var s = require(mod.module);
            mod.register(s);
            return true;
        } catch (ignored) {}
    }
});
```